File: Group_Anagrams/group_anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        words = {}
        for word in strs:
            ident = "".join(sorted(word))
            words.setdefault(ident, []).append(word)
        return[list for list in words.values()]
        

File: Group_Anagrams/group_anagrams.cpp
#include <algorithm>
#include <string>

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<std::string, std::vector<std::string>> words;
        for (int i = 0; i < (int)strs.size(); i++) {
            string word = strs[i];
            string ident = word;
            std::sort(ident.begin(), ident.end());
            words[ident].push_back(word);
        }
        std::vector<std::vector<std::string>> res;
        for (const auto&[key,group] : words) {
            res.push_back(group);
        }
        return res;
    }
};
